lib/qesapdeployment.pm

```diff
@@ -87,6 +91,56 @@ sub qesap_create_folder_tree {
     assert_script_run("mkdir -p $paths{deployment_dir}", quiet => 1);
 }
 
+=head3 qesap_get_variables
+
+    Scans yaml config for '%OPENQA_VARIABLE%' placeholders and searches for values in OpenQA defined variables.
+    Returns hash with openqa variable key/value pairs.
```

Comment:
Is it mostly intended to be used as input for **qesap_prepare_env**? Are there any other use cases for it?
If it is the only purpose, maybe we can change it to a private function and change  **qesap_prepare_env** to something like

```
sub qesap_prepare_env {
    my (%args) = @_;
    my $variables
    if($args{openqa_variables}) {
        $variables = $args{openqa_variables};
    }
    else {
        $variables = qesap_get_variables();
    }
```

Reply:
Yep that could work, I will make it like that. 
I just didn't want to force this mechanism into other functions.

Reply:
Change applied:
```
    my $variables = $args{openqa_variables} ? $args{openqa_variables} : qesap_get_variables();
```